client/src/main/scala/client/WebSocketClient.scala

```diff
@@ -0,0 +1,91 @@
+package client
+
+import client.WebSocketClient.WebSocketClientException
+import monifu.reactive.Ack.Continue
+import monifu.reactive.OverflowStrategy.DropOld
+import monifu.reactive._
+import monifu.reactive.channels.PublishChannel
+import org.scalajs.dom
+import org.scalajs.dom.raw.MessageEvent
+import org.scalajs.dom.{CloseEvent, ErrorEvent, Event, WebSocket}
+
+import scala.concurrent.Promise
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.util.{Success, Try}
+
+class WebSocketClient(url: String) extends Observable[String] with (String => Unit) {
+  private val sendOnOpen = Promise[String => Unit]()
+
+  def apply(message: String): Unit = {
```

Comment:
Does Scala treat apply in class somehow differently than apply function on object ?